iroh/src/commands.rs

```diff
@@ -55,66 +68,161 @@ pub struct Cli {
 }
 
 impl Cli {
-    pub async fn run(self, rt: &runtime::Handle, config: &Config) -> Result<()> {
+    pub async fn run(self, rt: &runtime::Handle) -> Result<()> {
         match self.command {
-            Commands::Share {
-                hash,
-                recursive,
-                rpc_port,
-                peer,
+            Commands::Console => {
+                let client = iroh::client::quic::connect_raw(self.rpc_args.rpc_port).await?;
+                let env = ConsoleEnv::for_console()?;
+                repl::run(client, env).await
+            }
+            Commands::Rpc(command) => {
+                let client = iroh::client::quic::connect_raw(self.rpc_args.rpc_port).await?;
+                let env = ConsoleEnv::for_cli()?;
+                command.run(client, env).await
+            }
+            Commands::Full(command) => {
+                let FullArgs {
+                    cfg,
+                    metrics_addr,
+                    keylog,
+                } = self.full_args;
+
+                let config = NodeConfig::from_env(cfg.as_deref())?;
+
+                #[cfg(feature = "metrics")]
+                let metrics_fut = start_metrics_server(metrics_addr, rt);
+
+                let res = command.run(rt, &config, keylog).await;
+
+                #[cfg(feature = "metrics")]
+                if let Some(metrics_fut) = metrics_fut {
+                    metrics_fut.abort();
+                }
+
+                res
+            }
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+pub enum Commands {
+    /// Start the Iroh console
+    Console,
+    #[clap(flatten)]
+    Full(#[clap(subcommand)] FullCommands),
+    #[clap(flatten)]
+    Rpc(#[clap(subcommands)] RpcCommands),
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Subcommand, Debug, Clone)]
+pub enum FullCommands {
+    /// Start a Iroh node
+    ///
+    /// If PATH is a folder all files in that folder will be served.  If no PATH is
+    /// specified reads from STDIN.
```

Comment:
what about not serving anything? which will kinda be the default for using `sync`

Reply:
Yes thought about that - I didn't change anything about the `start`  command apart from renaming from `provide` to `start` and adjusting docs a little. Maybe let's leave that for a followup.

Reply:
okay, lets have an issue about this though, as I think this is important to do sooner rather than later

Reply:
See https://github.com/n0-computer/iroh/issues/1403